emails/tests/models_tests.py

```diff
@@ -719,327 +564,292 @@ def test_save_server_storage_false_deletes_data(self):
+        social_account.extra_data["displayName"] = display_name
+        social_account.save()
+        assert self.profile.display_name == display_name
+
+    def test_display_name_does_not_exist(self) -> None:
+        self.get_or_create_social_account()
+        assert self.profile.display_name is None
+
 
-        for relay_address in RelayAddress.objects.filter(
-            user=server_stored_data_profile.user
-        ):
-            assert relay_address.description == test_desc
-            assert relay_address.generated_for == test_generated_for
+class ProfileLanguageTest(ProfileTestCase):
+    """Test Profile.language"""
 
-    def test_language_with_no_fxa_extra_data_locale_returns_default_en(self):
-        baker.make(SocialAccount, user=self.profile.user, provider="fxa")
+    def test_no_fxa_extra_data_locale_returns_default_en(self) -> None:
+        social_account = self.get_or_create_social_account()
+        assert "locale" not in social_account.extra_data
         assert self.profile.language == "en"
 
-    def test_language_with_no_fxa_locale_returns_default_en(self):
+    def test_no_fxa_locale_returns_default_en(self) -> None:
         assert self.profile.language == "en"
 
-    def test_language_with_fxa_locale_de_returns_de(self):
-        baker.make(
-            SocialAccount,
-            user=self.profile.user,
-            provider="fxa",
-            extra_data={"locale": "de,en-US;q=0.9,en;q=0.8"},
-        )
+    def test_fxa_locale_de_returns_de(self) -> None:
+        social_account = self.get_or_create_social_account()
+        social_account.extra_data["locale"] = "de,en-US;q=0.9,en;q=0.8"
+        social_account.save()
         assert self.profile.language == "de"
 
-    def test_locale_in_premium_country_returns_True_if_premium_available(self):
-        baker.make(
-            SocialAccount,
-            user=self.profile.user,
-            provider="fxa",
-            extra_data={"locale": "de-DE,en-xx;q=0.9,en;q=0.8"},
-        )
+
+class ProfileFxaLocaleInPremiumCountryTest(ProfileTestCase):
+    """Tests for Profile.fxa_locale_in_premium_country"""
+
+    def set_fxa_locale(self, locale: str) -> None:
+        social_account = self.get_or_create_social_account()
+        social_account.extra_data["locale"] = locale
+        social_account.save()
+
+    def test_when_premium_available_returns_True(self) -> None:
+        self.set_fxa_locale("de-DE,en-xx;q=0.9,en;q=0.8")
         assert self.profile.fxa_locale_in_premium_country is True
 
-    def test_locale_in_premium_country_returns_False_if_premium_unavailable(self):
-        baker.make(
-            SocialAccount,
-            user=self.profile.user,
-            provider="fxa",
-            extra_data={"locale": "en;q=0.8"},
-        )
+    def test_when_premium_unavailable_returns_False(self) -> None:
+        self.set_fxa_locale("en;q=0.8")
         assert self.profile.fxa_locale_in_premium_country is False
```

Comment:
question (blocking): why is premium unavailable in `en` locale in this test?

Reply:
That's fine. This PR isn't about cleaning up EVERY sloppy test, so yeah - let's land this nice re-organizing work.